query-primitives/src/main/scala/io/shiftleft/queryprimitives/utils/ExpandTo.scala

```diff
@@ -104,20 +101,22 @@ object ExpandTo {
       .toSeq
   }
 
-  def methodToTypeDecl(method: Vertex): Option[Vertex] = {
-    var typeDeclOption = method.vertices(Direction.IN, EdgeTypes.AST).asScala.toList.headOption
-    while (typeDeclOption.isDefined && !typeDeclOption.get.isInstanceOf[nodes.TypeDecl]) {
-      typeDeclOption = typeDeclOption.get.vertices(Direction.IN, EdgeTypes.AST).asScala.toList.headOption
+  @tailrec
+  def methodToTypeDecl(vertex: Vertex): Option[Vertex] = {
+    vertex.vertices(Direction.IN, EdgeTypes.AST).asScala.toList match {
```

Comment:
how about old skool - i find this easier to read. same applies to the one below (methodToFile):
```
val iter = vertex.vertices(Direction.IN, EdgeTypes.AST)
if (iter.hasNext) {
  iter.next match {
    case head: nodes.TypeDecl => Some(head)
    case head => methodToTypeDecl(head)
  }
} else None
```

Reply:
ah, yes, I was wondering how **not** to make it consume the whole list and forgot about iterators. Thanks!

Reply:
Done. Found a way to DRY even more